### DISCUSSION TITLE
GCC 11 warning -Wstringop-overflow (btAlignedObjectArray.h)

### DISCUSSION THREAD

**Discussion by issakomi** [bulletphysics/bullet3#3998] [Q&A]
Hello,

is any information available about the warning (multiple)? Linux (Fedora 34 x86_64), GCC 11. Never seen before, with other compilers/platforms.
Thank you.

```
/home/r/bullet/bullet3-master/src/LinearMath/btAlignedObjectArray.h:223:33: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’ writing between 8 and 17179869184 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  223 |                                 new (&m_data[i]) T(fillData);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Edit:**
Maybe related to [https://gcc.gnu.org/bugzilla/show_bug.cgi?id=92718](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=92718)?

**Comment by issakomi** [comment]
BTW, the warning disappears if

```
- new (&m_data[i]) T(fillData);
+ new (&m_data[(unsigned int)i]) T(fillData);
```

**Comment by erwincoumans** [comment]
Odd, how can i every become negative in that case?

**Comment by Levi-Armstrong** [comment]
I am getting the same build warning/error when building on Ubuntu Jammy in release mode.
